apps/language_server/lib/language_server/packet_router.ex

```diff
@@ -0,0 +1,92 @@
+defmodule ElixirLS.LanguageServer.PacketRouter do
+  defmodule State do
+    defstruct monitor_references: %{}
+
+    def new(names_or_pids) when is_list(names_or_pids) do
+      Enum.reduce(names_or_pids, %__MODULE__{}, &add(&2, &1))
+    end
+
+    def add(%__MODULE__{} = state, pid) when is_pid(pid) do
+      new_monitors = Map.put(state.monitor_references, Process.monitor(pid), pid)
+      %__MODULE__{state | monitor_references: new_monitors}
+    end
+
+    def add(%__MODULE__{} = state, name) do
+      add(state, Process.whereis(name))
+    end
+
+    def delete_ref(%__MODULE__{} = state, ref) when is_reference(ref) do
+      new_references = Map.pop(state.monitor_references, ref)
+      %__MODULE__{state | monitor_references: new_references}
+    end
+
+    def broadcast(%__MODULE__{} = state, type, message) do
+      broadcast_fn =
```

Comment:
`call` broadcasts will be blocking. Maybe use `Task.async_stream` to call concurrently?

Reply:
will this matter in practice?

Most of the new server's messages are handled via infos and casts and the ones that are called need to complete. They update its internal state and should be very quick to respond.
Adding an intermediate thing here seems like it would only add complexity and not gain much performance.

Reply:
OK, seems reasonable. I just wasn't sure what that will be used for.